src/main/java/com/google/sps/utility/GmailUtility.java

```diff
@@ -63,10 +65,21 @@ public static MessagePartHeader extractHeader(Message message, String headerName
    * @param fromHeaderValue the value of a "From" header from which a contact name / email should be
    *     extracted
    * @return A contact name if available, or an email if it is not
+   * @throws IllegalArgumentException If value could not be parsed from header
    */
   public static String parseNameInFromHeader(String fromHeaderValue) {
-    return fromHeaderValue.charAt(0) == '<'
-        ? fromHeaderValue.substring(1, fromHeaderValue.length() - 1)
-        : fromHeaderValue.split("<")[0].trim();
+    Pattern withNamePattern = Pattern.compile("(.*) <.*>");
```

Comment:
Make the Pattern a static variable. The `Pattern.compile` is a relatively expensive operation. If you have a constant pattern, it's best to only compile it once.

Reply:
Okay interesting! Good suggestion - thanks